cpp/src/arrow/compute/exec/exec_plan.cc

```diff
@@ -43,16 +44,20 @@
 namespace arrow {
 
 using internal::checked_cast;
+using internal::ThreadPool;
 using internal::ToChars;
 
 namespace compute {
 
 namespace {
 
 struct ExecPlanImpl : public ExecPlan {
-  explicit ExecPlanImpl(QueryOptions options, ExecContext* exec_context,
-                        std::shared_ptr<const KeyValueMetadata> metadata = NULLPTR)
-      : metadata_(std::move(metadata)), query_context_(options, *exec_context) {}
+  explicit ExecPlanImpl(QueryOptions options, ExecContext exec_context,
+                        std::shared_ptr<const KeyValueMetadata> metadata = NULLPTR,
+                        std::shared_ptr<ThreadPool> owned_thread_pool = nullptr)
```

Comment:
nit: be consistent with NULLPTR/nullptr (note that NULLPTR is really only necessary in headers, not in source files)

Reply:
Thanks, fixed!